presto-main/src/main/java/com/facebook/presto/cost/ComparisonStatsCalculator.java

```diff
@@ -0,0 +1,188 @@
+                        .setNullsFraction(0.0).build();
+
+        return inputStatistics.mapOutputRowCount(rowCount -> filterFactor * (1 - symbolStats.getNullsFraction()) * rowCount)
+                .mapSymbolColumnStatistics(symbol, oldStats -> symbolNewEstimate);
+    }
+
+    private PlanNodeStatsEstimate symbolToLiteralNonEquality(Symbol symbol, double literal)
+    {
+        SymbolStatsEstimate symbolStats = inputStatistics.getSymbolStatistics(symbol);
+
+        StatisticRange range = new StatisticRange(symbolStats.getLowValue(), symbolStats.getHighValue(), symbolStats.getDistinctValuesCount());
+        StatisticRange intersectRange = range.intersect(new StatisticRange(literal, literal, 1));
+
+        double filterFactor = 1 - range.overlapPercentWith(intersectRange);
+
+        return inputStatistics.mapOutputRowCount(rowCount -> filterFactor * (1 - symbolStats.getNullsFraction()) * rowCount)
+                .mapSymbolColumnStatistics(symbol, oldStats -> buildFrom(oldStats)
+                        .setNullsFraction(0.0)
+                        .setDistinctValuesCount(max(oldStats.getDistinctValuesCount() - 1, 0))
+                        .setAverageRowSize(oldStats.getAverageRowSize())
+                        .build());
+    }
+
+    private PlanNodeStatsEstimate symbolToLiteralEquality(Symbol symbol, double literal)
+    {
+        SymbolStatsEstimate symbolStats = inputStatistics.getSymbolStatistics(symbol);
+
+        StatisticRange range = new StatisticRange(symbolStats.getLowValue(), symbolStats.getHighValue(), symbolStats.getDistinctValuesCount());
+        StatisticRange intersectRange = range.intersect(new StatisticRange(literal, literal, 1));
+
+        double filterFactor = range.overlapPercentWith(intersectRange);
+        SymbolStatsEstimate symbolNewEstimate =
+                SymbolStatsEstimate.builder()
+                        .setAverageRowSize(symbolStats.getAverageRowSize())
+                        .setDistinctValuesCount(intersectRange.getDistinctValuesCount())
+                        .setHighValue(intersectRange.getHigh())
+                        .setLowValue(intersectRange.getLow())
+                        .setNullsFraction(0.0).build();
+
+        return inputStatistics.mapOutputRowCount(rowCount -> filterFactor * (1 - symbolStats.getNullsFraction()) * rowCount)
+                .mapSymbolColumnStatistics(symbol, oldStats -> symbolNewEstimate);
+    }
+
+    public PlanNodeStatsEstimate comparisonSymbolToSymbolStats(Symbol left, Symbol right, ComparisonExpressionType type)
+    {
+        switch (type) {
+            case EQUAL:
+                return symbolToSymbolEquality(left, right);
+            case NOT_EQUAL:
+                return symbolToSymbolNonEquality(left, right);
+            case LESS_THAN:
+            case LESS_THAN_OR_EQUAL:
+            case GREATER_THAN:
+            case GREATER_THAN_OR_EQUAL:
+            case IS_DISTINCT_FROM:
```

Comment:
No. Methods above are not capable of doing a = StatisticsRangeFrom(b), when b range is bigger the single point. We could add abstraction here but I see no point and we describe all those rules separately in doc anyway.